src/infra/ChatAnalyticsReporter.ts

```diff
@@ -0,0 +1,74 @@
+import type { EventPayload } from '../models/chat';
+import type ChatAnalyticsConfig from '../models/config/ChatAnalyticsConfig';
+
+interface EventsAPIResponse {
+  id: string;
+}
+
+/**
+ * A class to report chat analytics data. Uses the provided API key, environment,
+ * and partition to determine the endpoint for reporting events.
+ */
+export default class ChatAnalyticsReporter {
+  public readonly apiKey: string;
+  public readonly env: NonNullable<ChatAnalyticsConfig['env']>;
+  public readonly partition: NonNullable<ChatAnalyticsConfig['partition']>;
+  public readonly endpoint: string;
+
+  private endpoints: Record<
+    NonNullable<ChatAnalyticsConfig['partition']>,
+    Record<NonNullable<ChatAnalyticsConfig['env']>, string>
+  > = {
+    US: {
+      PROD: 'https://www.us.yextevents.com/accounts/me/events',
+      SANDBOX: 'https://www.sbx.us.yextevents.com/accounts/me/events',
+      QA: 'https://www.qa.us.yextevents.com/accounts/me/events',
+    },
+    EU: {
+      PROD: 'https://www.eu.yextevents.com/accounts/me/events',
+      SANDBOX: 'https://www.sbx.eu.yextevents.com/accounts/me/events',
```

Comment:
there is no SBX eu fyi.

Reply:
Oh I see. Should we throw an error if someone tries to do that?

Reply:
Will there ever be one?

Reply:
Pretty sure there will never be one. We're slowly moving towards getting rid of SBX even in the US partition.

Reply:
removed